src/SIL.XForge.Scripture/ClientApp/src/app/shared/text/text.component.ts

```diff
@@ -436,8 +436,32 @@ export class TextComponent extends SubscriptionDisposable implements OnDestroy {
     // As the browser is automatically applying ltr/rtl we need to ask it which one it is using
     // This value can then be used for other purposes i.e. CSS styles
     const quill = document.querySelector('quill-editor');
-    if (quill !== null) {
+    if (quill !== null && this.editor !== undefined) {
       this._direction = window.getComputedStyle(quill).direction;
+      // Direction can also applied to segments and paragraphs so check them as well
+      const elements = document.querySelectorAll('quill-editor *[dir=auto]');
+      if (elements !== null) {
+        for (const index in elements) {
```

Comment:
Why not `for ... of` rather than `for ... in` followed by the `hasOwnProperty` stuff?